src/ape/cli/options.py

```diff
@@ -112,12 +112,17 @@ def network_option(
         kwargs: Additional overrides to ``click.option``.
     """
 
-    if default is None and not required:
+    auto = default == "auto"
+
+    if auto and not required:
         if ecosystem:
             default = ecosystem[0] if isinstance(ecosystem, (list, tuple)) else ecosystem
         else:
             default = networks.default_ecosystem.name
 
+    elif auto:
+        default = None
```

Comment:
Allows the option to be `None`. Options are traditionally optional so it was weird that it would always come back `ethereum` even when I declare it to be `None`.

Reply:
Oh, I kind of like how it let's you run your default network, means I don't have to set it to anything

I think I prefer having the default be whatever you specify your default network/provider combo to be

Reply:
The default behavior does change here. It still uses your default ecosystem / network.
However, if you purposely do `default=None`, it won't do that. That is what this PR does.

Reply:
Let me show example of what I mean:

What is happening now:

```python
@network_option()
def cmd0(network):
  print(network)

@network_option(default=None)
def cmd1(network):
  print(network)
```

and you do `ape run cmd0  && ape run cm1` (no connections used, no NetworkBoundCommand in this problem), you get:

```
ethereum
ethereum
```

as the output.

But then if you use my PR, you get this at the output:

```
ethereum
None
```

which is more expected.
So by defaut, `default=auto` and it results in `ethereum`
But you can override that and say `default=None` to not have that behavior.
This **do not** have anythin to do with `NetworkBoundCommand` and probably would not work with `NetworkBoundCommand`. The use case is different.